### PR TITLE
PHOENIX-4763: Changing a base table property value should be reflected in child views (if the property wasn't changed)

### PR DESCRIPTION
**Pull request by ChinmaySKulkarni**
[WIP]
- When a view alters a property, add a tag for the corresponding cell in SYSCAT
- Introduce a bitset to keep track of these properties in PTable
- Add tests
- TODO:
1. Handle the case where we "CREATE VIEW <>" with properties specified
2. Add unit tests for tagging changes
3. Refactoring change for PHOENIX-4951 as part of this patch itself.

### CODE REVIEW

**Comment by ChinmaySKulkarni** [comment]
@twdsilva Please review. I will rebase changes once the approach looks good.

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java [hunk 1]
```diff
@@ -2340,7 +2365,7 @@ public void createTable(RpcController controller, CreateTableRequest request,
                 // 3. Finally write the mutations to create the table
 
                 // From 4.15 the parent->child links are stored in a separate table SYSTEM.CHILD_LINK
-                // TODO remove this after PHOENIX-4763 is implemented
+                // TODO remove this after PHOENIX-4763 is implemented -> Remove below?
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java]: @twdsilva Is this comment just to keep track of the fact that we need to remove this after all of splittable SYSCAT is implemented or is there anything specific to PHOENIX-4763?

**Comment by ChinmaySKulkarni** [comment]
- Added cell tagging change in 'CREATE VIEW' with properties path
- Added tests
- Added refactoring as per PHOENIX-4951
@twdsilva please review. Thanks!

File: phoenix-core/src/main/java/org/apache/phoenix/util/MetaDataUtil.java [hunk 2]
```diff
@@ -353,6 +403,18 @@ public static int getSaltBuckets(List<Mutation> tableMetaData, KeyValueBuilder b
         }
         return 0;
     }
+
```
Comment (twdsilva) [inline phoenix-core/src/main/java/org/apache/phoenix/util/MetaDataUtil.java]: remove these two functions if they aren't used

File: phoenix-core/src/test/java/org/apache/phoenix/util/MetaDataUtilTest.java [hunk 3]
```diff
@@ -65,21 +100,67 @@ public void testCompatibility() {
...
-        assertEquals(Bytes.toString(newValue), Bytes.toString(cell.getValueArray(), cell.getValueOffset(), cell.getValueLength()));
+        Cell cell = put.get(TABLE_FAMILY_BYTES, QUALIFIER).get(0);
+        assertEquals(Bytes.toString(ORIGINAL_VALUE),
+                Bytes.toString(cell.getValueArray(), cell.getValueOffset(), cell.getValueLength()));
+        MetaDataUtil.mutatePutValue(put, TABLE_FAMILY_BYTES, QUALIFIER, newValue);
+        cell = put.get(TABLE_FAMILY_BYTES, QUALIFIER).get(0);
+        assertEquals(Bytes.toString(newValue),
+                Bytes.toString(cell.getValueArray(), cell.getValueOffset(), cell.getValueLength()));
+    }
+
```
Comment (twdsilva) [inline phoenix-core/src/test/java/org/apache/phoenix/util/MetaDataUtilTest.java]: Nice unit tests.

File: phoenix-core/src/main/java/org/apache/phoenix/util/MetaDataUtil.java (hunk 2 above)
Comment (ChinmaySKulkarni) [reply phoenix-core/src/main/java/org/apache/phoenix/util/MetaDataUtil.java]: good catch. Will remove

**Comment by ChinmaySKulkarni** [comment]
Thanks for the review @twdsilva

**Pull request closed without merging**